clusterloader2/pkg/measurement/common/slos/pod_startup_latency.go

```diff
@@ -128,15 +137,38 @@ func (p *podStartupLatencyMeasurement) start(c clientset.Interface) error {
 				return c.CoreV1().Pods(p.selector.Namespace).Watch(context.TODO(), options)
 			},
 		},
-		p.checkPod,
+		p.addEvent,
 	)
+	go p.processEvents()
 	return informer.StartAndSync(i, p.stopCh, informerSyncTimeout)
 }
 
+func (p *podStartupLatencyMeasurement) addEvent(_, obj interface{}) {
+	event := &eventData{obj, time.Now()}
```

Comment:
Let's set field names explicitly (otherwise it becomes harder to refactor later when we add new field and not set all).